File: leetcode/1295_find_numbers_with_even_number_of_digits.py
# -*- coding: utf-8 -*-

from typing import List


class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        result = 0
        for num in nums:
            if 10 <= num < 100 or 1000 <= num < 10000:
                result += 1
        return result


if __name__ == '__main__':
    solution = Solution()

    assert 2 == solution.findNumbers([12, 345, 2, 6, 7896])
    assert 1 == solution.findNumbers([555, 901, 482, 1771])

File: leetcode/1295_find_numbers_with_even_number_of_digits.cpp
// Copyright (c) 2020 Jacopo Notarstefano

#include <algorithm>
#include <cassert>
#include <vector>

using namespace std;

class Solution {
 public:
    int findNumbers(vector<int>& nums) {
        return count_if(nums.begin(), nums.end(), evenDigits);
    }

 private:
    static bool evenDigits(int num) {
        return (10 <= num && num < 100) || (1000 <= num && num < 10000);
    }
};

int main() {
    auto solution = Solution();

    vector<int> nums = {12, 345, 2, 6, 7896};

    assert(2 == solution.findNumbers(nums));
}
